app/utils/rateLimit.js

```diff
@@ -0,0 +1,67 @@
+
+const Redis = require('ioredis')
+const { RateLimiterRedis } = require('rate-limiter-flexible')
+
+const redisClient = new Redis({
+  port: process.env.REDIS_PORT,
+  host: process.env.REDIS_HOST,
+  family: 4,
+  password: process.env.REDIS_PASSWORD,
+  db: process.env.REDIS_DB
+})
+
+const loginLimiter = new RateLimiterRedis({
+  storeClient: redisClient,
+  keyPrefix: 'login',
+  points: 5,
+  duration: 60 * 30 // 30 minutes wait time
+})
```

Comment:
@kmehant You are rate-limiting only for the login endpoints so DOS attack is still possible by a logged-in user.
Let say I am a logged in user and now I will write a script something like this:
```
while(1) {
    setInterval(() => { // hit endpoint to send GET/POST request to the server }, 1000);
}
```
This will hit the server after every 1second and due to infinite hit server will go down.

Please take it into consideration.
cc @vaibhavdaren

Reply:
@Rupeshiya Yes, this point is noted ✌🏻.
I will create another issue and add these points there and will work on another PR as this change is especially required for login endpoint 😅 to create time block delays.
But the present work is to create restrict **unsuccessful** login attempts that come under rate-limiting but has a special extension of using block time intervals. (was the reason for not writing this as a middleware rather an extension to login controller)

So in our next pr, I will simply write a middleware that will keep track of the number of requests (immediate) and IP address of the Client, and then we will use a quota system where we will block the user from accessing the service. (this truly comes under rate-limiting 😅 and not the present work, sorry if  I have used the word rate-limiting anywhere in the context for the present work.)

Reply:
Yes!!
Great go ahead @kmehant !!